osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -623,6 +612,26 @@ protected override Track GetTrack()
             }
 
             protected override Waveform GetWaveform() => new Waveform(store.GetStream(getPathForFile(Metadata.AudioFile)));
+
+            protected override Storyboard GetStoryboard()
+            {
+                if (BeatmapSetInfo?.StoryboardFile == null)
+                    return new Storyboard();
+
+                try
+                {
+                    BeatmapDecoder decoder;
+                    using (var stream = new StreamReader(store.GetStream(getPathForFile(BeatmapInfo.Path))))
+                        decoder = BeatmapDecoder.GetDecoder(stream);
+

```

Comment:
I think it might still be desirable to parse the beatmap file with `DecodeStoryboard`, for the cases where the storyboard is not solely inside the .osb. I found the following map which does not have the storyboard in an osb: https://osu.ppy.sh/beatmapsets/797#osu/5089. Thoughts, @peppy, or should we fix these on a per-case basis?

Reply:
I added something for that. If there is no .osb then it uses the .osu. If it's mixed it's a different thing though...